Library/Homebrew/cask/cask_loader.rb

```diff
@@ -236,7 +236,9 @@ def self.for(ref)
       when 2..Float::INFINITY
         loaders = possible_tap_casks.map(&FromTapPathLoader.method(:new))
 
-        raise TapCaskAmbiguityError.new(ref, loaders)
+        raise TapCaskAmbiguityError.new(ref, loaders) if loaders.map(&:tap).map(&:name).uniq.size == 1
```

Comment:
Sorry, still don't understand what `loaders.map(&:tap).map(&:name).uniq.size` is doing here, can you explain?

Reply:
@MikeMcQuaid 
The bug in https://github.com/Homebrew/brew/issues/13628 happens because `(possible_tap_casks = tap_paths(ref)).count` is 2 at https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/cask_loader.rb#L233 without checking the tap's full name in the following flow. so I repair to check loaders's tap name is uniq or not. 
However this PR's approace may be wrong.

1. https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cmd/info.rb#L232
```
"casks"    => casks.map(&:to_h),
```

2. https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/cask.rb#L23
```
 "versions"       => os_versions,
```

3. https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/cask.rb#L99
```
cask = CaskLoader.load(token)
```

4. https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/cask_loader.rb#L233
```
case (possible_tap_casks = tap_paths(ref)).count
```